# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1260315640 18000
#      Tue Dec 08 18:40:40 2009 -0500
# Node ID fea3d4868692ba2256cdaa47285f09275d1706af
# Parent  4c681ff57bc379da6776b264cd4d2fdcad7002e7
__go_draw_figure__.m: Add missing 'unset multiplot' to restore mouse actions.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-12-08  Petr Mikulik <mikulik@physics.muni.cz>
+
+	* plot/__go_draw_figure__.m: Add missing 'unset multiplot' to restore
+	hotkey and mouse functionality for 2D plots.
+
 2009-12-09  Rik <octave@nomad.inbox5.com>
 
 	* Makefile.am: Distribute DOCSTRINGS in tar.gz file
 	* mkdoc: pass current directory, "./", when running mkdoc on local files
 
 2009-12-07  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/interp1q.m: Use more optimal code.
diff --git a/scripts/plot/__go_draw_figure__.m b/scripts/plot/__go_draw_figure__.m
--- a/scripts/plot/__go_draw_figure__.m
+++ b/scripts/plot/__go_draw_figure__.m
@@ -90,16 +90,17 @@ function __go_draw_figure__ (h, plot_str
 		unwind_protect_cleanup
 		set (kids(i), "units", orig_axes_units);
 		set (kids(i), "position", orig_axes_position);
 	      end_unwind_protect
 	    otherwise
 	      error ("__go_draw_figure__: unknown object class, %s", type);
 	  endswitch
 	endfor
+	fputs (plot_stream, "\nunset multiplot;\n");
       else
 	fputs (plot_stream, "\nreset; clear;\n");
 	fflush (plot_stream);
       endif
     else
       error ("__go_draw_figure__: expecting figure object, found `%s'",
 	     htype);
     endif
